### PR TITLE
[processor/groupbyattrs] move from OpenCensus to OpenTelemetry

### LINKED ISSUE #30763: processor/groupbyattrsprocessor

### PR DESCRIPTION
**Pull request by codeboten**
Fixes #30763

### CODE REVIEW

File: processor/groupbyattrsprocessor/metrics.go [hunk 1]
```diff
@@ -4,87 +4,123 @@
...
+		metric.WithDescription("Number of spans that did not have attributes grouped"),
+		metric.WithUnit("1"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	it.mNumNonGroupedSpans = counter
+
+	histo, err := metadata.Meter(set).Int64Histogram(
```
[the remark this answers is not in the document]
Comment (codeboten) [reply processor/groupbyattrsprocessor/metrics.go:52]: i don't have a strong preference here, the codebase already has a mix of both `histo` and `histogram`

File: processor/filterprocessor/telemetry_test.go
Comment (TylerHelmuth) [inline processor/filterprocessor/telemetry_test.go:1]: Did you handle this file in a separate PR?

Comment (codeboten) [reply processor/filterprocessor/telemetry_test.go:1]: I did not, but I can. I'm just renaming the import here

File: processor/groupbyattrsprocessor/metrics.go [hunk 2]
```diff
@@ -4,87 +4,123 @@
...
-	mDistLogGroups     = stats.Int64("log_groups", "Distribution of groups extracted for logs", stats.UnitDimensionless)
+	mNumGroupedLogs    metric.Int64Counter
+	mNumNonGroupedLogs metric.Int64Counter
+	mDistLogGroups     metric.Int64Histogram
 
-	mNumGroupedMetrics    = stats.Int64("num_grouped_metrics", "Number of metrics that had attributes grouped", stats.UnitDimensionless)
-	mNumNonGroupedMetrics = stats.Int64("num_non_grouped_metrics", "Number of metrics that did not have attributes grouped", stats.UnitDimensionless)
-	mDistMetricGroups     = stats.Int64("metric_groups", "Distribution of groups extracted for metrics", stats.UnitDimensionless)
-)
+	mNumGroupedMetrics    metric.Int64Counter
```
[the remark this answers is not in the document]
Comment (codeboten) [reply processor/groupbyattrsprocessor/metrics.go:23]: I figured a monotonic counter was the right instrument to use here as I expect this number to only ever go in one direction. I'm not sure if there's a clear translation map somewhere for OpenCensus to OpenTelemetry metrics

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**